### DISCUSSION TITLE
Прошивка JL_A9C

### DISCUSSION THREAD

**Discussion by anbbox** [Arkady23/V380-Web#24] [Q&A]
Через wireshark ловится:

http://ipcupdate.av380.net/?param=eyJzb2Z0X2lkIjoiSkxfQTlDIiwiZGV2X2lkIjo1MjEwNjI4OCwidmVyc2lvbiI6MTAwMDkwMCwidGltZXN0YW1wIjoxNTkwOTcwMDg5fQ==

{"soft_id":"JL_A9C","dev_id":52106288,"version":1000900,"timestamp":1590970089}

Пробовал установщик httpd_JL_A9C_9090909_v20210620 - ничего не происходит, камера игнорирует файлы на sdcard.

Если закинуть на карточку ceshi.ini на включение rtsp - камера сбрасывается, через приложение определяется как ненастроенная. Порты при этом не открываются.

Есть идеи, как включить rtsp или прошить ее?

Алексей.

**Comment by Arkady23** [comment]
Были такие случаи, помогало указать версию на 1 больше, чем ваша текущая версия. Я в течение суток подготовлю вам версию для JL_A9C 1000900+1

**Comment by Arkady23** [comment]
Попробуйте версию патча 1000901.
https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_JL_A9C_1000901_v20210620.zip

**Comment by anbbox** [comment]
С этой версией патча картина та же самая - ничего не происходит. Изначально копирую содержимое архива на свежеформатированную карту. После включения камеры и ожидания около 5 минут - результат нулевой. На карте появляется каталог RECS для хранения видеозаписей - это как бы говорит о том, что карту камера видит и на нее пишет. Но патч, видимо, игнорирует. Вот скрин содержимого карты ПОСЛЕ попытки пропатчить.

![image](https://user-images.githubusercontent.com/26741783/150000282-7a44f60f-f18e-48fe-9a2d-628961ee0845.png)

Есть идеи, что еще можно попробовать? Подключиться к serial порту не смогу, нет оборудования и у меня лапки :(

**Reply by Arkady23** [reply]
Последнее время появляются новые версии оборудования, тем не менее пользователи обращаются в поддержку (например [здесь](https://github.com/Arkady23/V380-Web/discussions/21)) и им присылают последний патч к их камере. В этом случае можно исследовать этот патч и с его помощью сделать свой.

**Comment by Arkady23** [comment]
Выяснилось на другой модели камеры, что разработчик немного поменял логику обновлений. Возможно из-за этого старые патчи не срабатывают. Внёс поправку в патч. Попробуйте [такой вариант](https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_JL_A9C_9090909_v20210620.zip) https://github.com/Arkady23/V380-Web/releases/download/20210530/httpd_JL_A9C_9090909_v20210620.zip

**Comment by anbbox** [comment]
с этой версией у меня поведение камеры точно такое же, обновления не происходит. ссылка точно верная? контрольная сумма у скачанного архива такая же, как и ранее.

**Comment by Arkady23** [comment]
Сообщите сведения о вашей камере, отображаемые в пункте меню Версия ПО. Мне сейчас пришла такая же в ней указана версия прошивки HwV380E12_FW3_AK_MINI_20180628. Предполагаю, что у вас такая же камера. Я патч уже установил согласно таблице. Всё работает.

**Comment by anbbox** [comment]
Скрин
![IMG_20220213_112645](https://user-images.githubusercontent.com/26741783/153745287-f5a9de7a-cd10-4f4a-997a-8e2ef6aefdc4.jpg)

**Reply by Arkady23** [reply]
Странно. Я специально заказал вашу камеру https://aliexpress.ru/item/4001089979397.html?_ga=2.169380065.1149054730.1644656649-910746486.1590901087&sku_id=10000014304967618&spm=a2g39.orderlist.0.0.51a94aa6mJAd1g Пришла с другой прошивкой:  HwV380E12_WF3_AK_MINI. Качество видео ужасное.

**Comment by anbbox** [comment]
выложил дамп трафика с камеры при включении [https://github.com/anbbox/V380-Web/blob/main/camera.zip](url)

**Reply by Arkady23** [reply]
А зачем при включении? Нужен момент от нажатия на пункт "Версия прошивки" и до момента отображения информации, что у вас последняя прошивка.

**Reply by anbbox** [reply]
https://github.com/anbbox/V380-Web/blob/8554e3c4ddee96f4bd8caa1704574b9c3faafc06/camera2.zip

**Comment by Arkady23** [comment]
Да, в самом начале вы тот же param нашли, что и сейчас. Поэтому даже и не знаю, как быть. Может это какая-то очень старая камера или наоборот очень новая? На всякий случай сделал такой вариант
[httpd_JL_A9C_1000910_v20210620.zip](https://github.com/Arkady23/V380-Web/files/8064533/httpd_JL_A9C_1000910_v20210620.zip)
Если не подойдёт, то надежда только, если вы получите какой-нибудь официальный патч к своей камере.
Sd карта в Fat32? Sd карта не сбойная?
Еще есть подозрение, что возможно ошибка в Hwname. Ведь Hwname всех камер кроме вашей начинается с V380. Может ваш Hwname не JL_A9C, а V380JL_A9C?

**Reply by anbbox** [reply]
с данным вариантом все то же самое. камера карту видит, т.к. на ней появляется каталог RECS и avi записи, но на прошивку не реагирует, скрины я прикладывал выше https://user-images.githubusercontent.com/26741783/150000282-7a44f60f-f18e-48fe-9a2d-628961ee0845.png

Написал в поддержку просьбу выслать прошивку, посмотрим, что ответят.

В любом случае спасибо за помощь.

**Reply by Arkady23** [reply]
Мне пришла такая камера. Приложение V380-pro работает не в полном объеме. Например, совсем не работает режим записи по движению, только запись постоянно. Сразу не обратил внимания, что у вас вместо папки RecFiles создается папка RECS. Ну и внутри этой папки всё по-другому. Я думаю, что и ПО другое. Поэтому мой web-сервер, даже если его установить, работать не должен. Для этой камеры надо делать другой web-сервер. Возможно, китайцы переделывают ПО, оно еще сырое. Думаю это может быть связано с перегрузкой слабого процессора на этой камере. У меня есть внешне такая же камера, но с другой системой, на которую http-сервер встаёт, но работает с зависаниями. Пришлось его отключить. Т.е. в режиме настроек, можно что-то изменить, но для работы я его вынужден отключать. Мне кажется эта камера очень слабая и плохого качества. Лучше их не покупать. У меня таких 2 с разными прошивками. Одна может записывать по движению, а вторая пока нет.
